src/cr/cube/cube.py

```diff
@@ -410,16 +414,23 @@ def _ca_as_0th(self):
     @lazyproperty
     def _cube_dict(self):
         """dict containing raw cube response, parsed from JSON payload."""
+        cube_dict = copy.deepcopy(self._cube_response)
+        if self._mean_subvariables:
+            dimensions = cube_dict.get("result", {}).get("dimensions", {})
+            dimensions.append(self._numeric_array_dimension)
+        return cube_dict
+
+    @lazyproperty
+    def _cube_response(self):
+        """dict representing the parsed cube response arguments."""
         try:
-            cube_response = self._cube_response_arg
+            response = self._cube_response_arg
             # ---parse JSON to a dict when constructed with JSON---
-            cube_dict = (
-                cube_response
-                if isinstance(cube_response, dict)
-                else json.loads(cube_response)
+            cube_response = (
+                response if isinstance(response, dict) else json.loads(response)
             )
             # ---cube is 'value' item in a shoji response---
-            return cube_dict.get("value", cube_dict)
+            return cube_response.get("value", cube_response)
```

Comment:
Yeah, some validation definitely wouldn't hurt here, but maybe best to leave it for later (because it needs to be a proper validator class, and not a bunch of ifs thrown around).

Reply:
???